scripts/generate_spiders.py

```diff
@@ -0,0 +1,39 @@
+import argparse
+import csv
+from jinja2 import Environment, FileSystemLoader
+
+
+def load_template(spider_data):
+    env = Environment(loader=FileSystemLoader("."))
+    template = env.get_template("spider_template.py")
+    print(
+        template.render(
+            spider_class_base=spider_data["spider_class_base"],
+            spider_class_name=spider_data["spider_class_name"],
+            spider_name=spider_data["spider_name"],
+            allowed_domain=spider_data["allowed_domain"],
+            start_date=spider_data["start_date"],
+            base_url=spider_data["base_url"],
+            territory_id=spider_data["territory_id"],
+        )
+    )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser("Generate a spider structure from CSV.")
+    parser.add_argument("file")
+    args = parser.parse_args()
+
+    with open(args.file) as csv_file:
+        reader = csv.DictReader(csv_file)
+        for row in reader:
+            spider_data = {
+                "spider_class_base": row["base"],
+                "spider_class_name": f"{row['state'].title()}{row['city'].title()}",
+                "spider_name": f"{row['state'].lower()}_{row['city'].lower()}",
+                "allowed_domain": row["domain"],
+                "start_date": "",
+                "base_url": row["base_url"],
+                "territory_id": row["territory_id"],
+            }
+            load_template(spider_data)

```

Comment:
The suggestions are simple, but as it changes most parts of the script, it became a suggestion for the whole file :grimacing:

Basically, I suggest a way to deal with multi-word cities, non-ascii characters and also export the content to the respective files.
```suggestion
import argparse
import csv
from os.path import abspath, dirname
from unicodedata import normalize

from jinja2 import Environment, FileSystemLoader


def load_template(spider_data):
    env = Environment(loader=FileSystemLoader("."))
    template = env.get_template(f"spider_template.py")
    return template.render(
        spider_class_base=spider_data["spider_class_base"],
        spider_class_name=spider_data["spider_class_name"],
        spider_name=spider_data["spider_name"],
        allowed_domain=spider_data["allowed_domain"],
        start_date=spider_data["start_date"],
        base_url=spider_data["base_url"],
        territory_id=spider_data["territory_id"],
    )


if __name__ == "__main__":
    parser = argparse.ArgumentParser("Generate a spider structure from CSV.")
    parser.add_argument("file")
    args = parser.parse_args()

    project_root = dirname(dirname(abspath(__file__)))

    with open(args.file) as csv_file:
        reader = csv.DictReader(csv_file)
        for row in reader:
        	normalized_city_name = (
                normalize("NFKD", row["city"]).encode("ASCII", "ignore").decode("utf8")
            )
            split_city_name = normalized_city_name.replace("-", " ").split()

            spider_data = {
                "spider_class_base": row["base"],
                "spider_class_name": f"{row['state'].title()}{''.join(split_city_name).title()}",
                "spider_name": f"{row['state'].lower()}_{'_'.join(split_city_name).lower()}",
                "allowed_domain": row["domain"],
                "start_date": "",
                "base_url": row["base_url"],
                "territory_id": row["territory_id"],
            }

            filepath = f"{project_root}/data_collection/gazette/spiders/{row['state'].lower()}_{'_'.join(split_city_name).lower()}.py"

            with open(filepath, "w") as spiderfile:
                spiderfile.write(load_template(spider_data))
```

Reply:
Thanks for your suggestions, @giuliocc! I've implemented most of it. :) Works like a charm.